src/components/grader/LabSelector.js

```diff
@@ -17,6 +17,7 @@ function LabSelector(props) {
     apiClient.getLabs(props.course).then(res => {
```

Comment:
I have a small recommendation, I guess it's late :sweat_smile: but I'll type it. Consider wrapping the API client with a service object that will unwrap the return value of Axios.This service object is responsible for taking status code and data objects from responses and mapping them to business objects.

For example

```javascript
class GradingService {
 getGrades(courseName) {
  let { status, data } = apiClient.getGrades(courseName);
  if (status != 200) {
    return {
      isSuccess: false,
      errors: data.errors // the API will always send an errors array when there's a problem
    }
  }

  return {
   isSuccess: true,
   grades: data.grades
  }
 }
}

Reply:
https://github.com/CC451-Computer-Networks-TA-team/GraderX-FE/issues/44